cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -359,7 +362,55 @@ private void restartIfNecessary(int podId, RestartContext restartContext)
             } else {
                 throw e;
             }
+        } catch (IllegalArgumentException e) {
+            if (isPodStuck(pod) || restartContext.backOff.done()) {
+                if (canRoll(podId, 60_000, TimeUnit.MILLISECONDS, true)) {
+                    log.warn("{}: Pod {} will be force-rolled, due to error: {}", reconciliation, podName(podId), e.getCause() != null ? e.getCause().getMessage() : e.getMessage());
+                    restartAndAwaitReadiness(pod, operationTimeoutMs, TimeUnit.MILLISECONDS);
+                } else {
+                    log.warn("{}: Pod {} can't be safely force-rolled; original error: ", reconciliation, podName(podId), e.getCause() != null ? e.getCause().getMessage() : e.getMessage());
+                    throw e;
+                }
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    private boolean isCrashLooping(Pod pod) {
+        return podWaitingBecauseOfReason(pod, "CrashLoopBackOff");
+    }
+
+    private boolean isImagePullBackOff(Pod pod) {
+        return podWaitingBecauseOfReason(pod, "ImagePullBackOff");
+    }
+
+    private boolean isCreating(Pod pod) {
+        return podWaitingBecauseOfReason(pod, "ContainerCreating");
+    }
+
+    private boolean podWaitingBecauseOfReason(Pod pod, String reason) {
+        if (pod != null && pod.getStatus() != null) {
+            List<ContainerStatus> kafkaContainerStatus = pod.getStatus().getContainerStatuses().stream().filter(containerStatus -> containerStatus.getName().equals("kafka")).collect(Collectors.toList());
+            if (kafkaContainerStatus.size() > 0) {
```

Comment:
If you're just interested in getting the first reason you should use `findFirst`, rather than `collect`ing into a `List`. But is it correct that the container status we're looking for will always be the first in the list?

Reply:
from the tests, yes

Reply:
That doesn't really mean that there does not exist a real-world circumstance where there's some other status which appears first, just that we've not run into such a circumstance yet.

Reply:
What do you suggest?

Reply:
I guess you can go through the list and check if it contains the conditions instead of checking just the first one?